packages/myst-config/src/errorRules/validators.ts

```diff
@@ -0,0 +1,63 @@
+import type { ValidationOptions } from 'simple-validators';
+import {
+  defined,
+  incrementOptions,
+  validateChoice,
+  validateList,
+  validateObjectKeys,
+  validateString,
+} from 'simple-validators';
+import type { ErrorRule } from './types.js';
+
+const ERROR_RULE_KEY_OBJECT = {
+  required: ['id'],
+  optional: ['severity'],
+  alias: {
+    rule: 'id',
+  },
+};
+
+export function validateErrorRule(input: any, opts: ValidationOptions): ErrorRule | undefined {
+  if (typeof input === 'string') {
+    input = { id: input };
+  }
+  const value = validateObjectKeys(input, ERROR_RULE_KEY_OBJECT, {
+    ...opts,
+    suppressWarnings: true,
+    keepExtraKeys: true,
+  });
+  if (value === undefined) return undefined;
+  const id = validateString(value.id, incrementOptions('id', opts));
+  if (!id) return undefined;
+  const output: ErrorRule = { id };
+  if (defined(value.severity)) {
```

Comment:
Feels a little ambiguous about what happens when only `id` is specified. Looks like it remains `undefined` - then, in `addWarningForFile` the default will be replaced by `undefined` and hit the `default` case in the switch statement... Hmm, then in the store, we will also get an undefined warning `kind`.

I think we need to make `severity` required on the type, and set a default here during validation?

Reply:
That begs the question what the default is... If I put `error_rule: my-id` in my config... am I expecting that will suppress it...? Based on the name "error rule" I'd more expect the default behaviour to be "Error on rule my-id" - so the default `severity` is `'error'`?

But then what's the use-case? Are we intending this for error suppression as the default behaviour? If so, maybe we need to rethink `error_rule`...?